app/code/local/Magmodules/Webwinkelconnect/Helper/Data.php

```diff
@@ -97,7 +97,133 @@ public function isProductReviewInviteEnabled(): bool {
+                ->setStoreId(Mage::app()->getStore()->getStoreId())
+                ->setStores([Mage::app()->getStore()->getStoreId()])
+                ->setCustomerId($this->getCustomerId($product_review['product_review']['reviewer']['email']))
+                ->setNickname($product_review['product_review']['reviewer']['name'])
+                ->save();
+
+            $rating_option_id = Mage::getModel('rating/rating_option')->getCollection()
+                ->addFieldToFilter('rating_id', Mage::getStoreConfig('webwinkelconnect/product_review_invites/rating'))
+                ->addFieldToFilter('code', $product_review['product_review']['rating'])
+                ->getFirstItem()
+                ->getOptionId();
+
+            Mage::getModel('rating/rating')
+                ->setRatingId(Mage::getStoreConfig('webwinkelconnect/product_review_invites/rating'))
+                ->setReviewId($review->getId())
+                ->addOptionVote($rating_option_id, $product_id);
+            $review->aggregate();
+            $review->setCreatedAt($product_review['product_review']['created']);
+            $review->save();
+
+            $connection->commit();
+            return [
+                'message' => json_encode(['review_id' => $review->getId()]),
+                'code' => 200
+            ];
+        } catch (Exception $e) {
+            $connection->rollback();
+            return ['message' => $e->getMessage(), 'code' => 500];
+        }
+    }
+
+
+    public function hasConsent(Mage_Sales_Model_Order $order): bool {
+        $permission_url = 'https://dashboard.webwinkelkeur.nl/api/2.0/order_permissions.json?' .
+            http_build_query([
+                'orderNumber' => $order->getIncrementId(),
+                'id' => $this->getShopId(),
+                'code' => $this->getApiKey(),
+            ]);
+
+        $start_time = microtime(true);
+        $ch = $this->getCurlHandle($permission_url);
+
+        try {
+            $response = curl_exec($ch);
+            if ($response === false) {
+                $message = sprintf("Consent check failed with cURL error: (%s) %s", curl_errno($ch), curl_error($ch));
+            } elseif (empty(json_decode($response)->has_consent)) {
+                $message = 'Invitation has not been sent as the customer did not consent.';
+            } else {
+                return true;
+            }
+
+            Mage::getModel('webwinkelconnect/log')->addToLog(
+                'invitation',
+                $order->getStoreId(),
+                '',
+                $message,
+                (microtime(true) - $start_time),
+                'orderupdate',
+                $permission_url,
+                $order->getId()
+            );
+            return false;
+        } finally {
+            curl_close($ch);
+        }
+    }
+
+    public function getCurlHandle(string $url, array $options = []) {
+        $ch = curl_init();
```

Comment:
If it's possible to save multiple orders or create multiple shipments in one go, and thereby trigger multiple calls to the API in a single request, you should reuse the cURL handle.

In any case, there'll be two requests when the consent is turned on, one for the consent, and one for sending the invite. If you use two separate cURL handles that will mean building up two TLS connections, which actually takes a lot of roundtrips (and time).

Better use a single cURL handle stored as a object property on this helper, that is initialized the first time you need it, and that is reset every time using `curl_reset()`. This will allow you to use different options for each request, but reuse the connection.

That means you can also drop the `try { ... } finally { curl_close($ch); }` stuff.